rbac/api/cross_access/inventory_api_dual_write_cross_access_handler.py

```
@@ -0,0 +1,327 @@
+        user_id = self._user_id()
+        source_key = self._source_key()
+
+        for role in roles:
+            for scope in self._resource_service.binding_scopes_for_role(role):
+                self._update_mapping_for_system_role(
+                    role,
+                    scope=scope,
+                    update_mapping=add_principal_to_binding,
+                    create_default_mapping_for_system_role=(
+                        lambda resource: self._create_default_mapping_for_system_role(
+                            system_role=role,
```

Ruff (0.16.1)

[warning] Function definition does not bind loop variable `role`

(B023)